yt/yt/server/query_tracker/handler_base.cpp

```diff
@@ -342,7 +347,11 @@ void TQueryHandlerBase::TryWriteProgress()
 bool TQueryHandlerBase::TryWriteQueryState(EQueryState state, EQueryState previousState, const TError& error, const std::vector<TErrorOr<TWireRowset>>& wireRowsetOrErrors)
 {
     try {
-        auto transaction = StartIncarnationTransaction(previousState);
+        ITransactionPtr transaction;
+        TActiveQuery record;
+        std::tie(transaction, record) = StartIncarnationTransaction(previousState);
+
+        auto now = TInstant::Now();
```

Comment:
Would be nice to tie all these timestams to actual transaction timestamps that change state

Reply:
Why do we want this?

Now it is consistent with other times (like start_time, finish_time)
https://github.com/ytsaurus/ytsaurus/blob/main/yt/yt/server/query_tracker/query_tracker_proxy.cpp#L682

And also transaction timestamps are less precise. (seconds instead of microseconds)
https://github.com/ytsaurus/ytsaurus/blob/main/yt/yt/client/transaction_client/public.h#L67